### PR TITLE
RUBY 2477 wcr tech upgrade to rails 7

### PR DESCRIPTION
**Pull request by jjromeo**
- RUBY-2477 Upgrade ruby version
- Unpin gems in gemfile and bundle update

### CODE REVIEW

**Review by PaulDoyle-DEFRA — DISMISSED** [review]

File: Gemfile.lock [hunk 1]
```diff
@@ -1,25 +1,35 @@
+GIT
+  remote: https://github.com/DEFRA/defra-ruby-mocks.git
+  revision: 8056a090711c1339e1265efc5d91552d7d549040
```
Comment (jjromeo) [inline Gemfile.lock:3]:
```suggestion
  revision: 52db1b539241b863ef4ddd36f335dc1901b9fef5
```

File: Gemfile [hunk 2]
```diff
@@ -51,16 +51,15 @@ gem "net-pop"
 gem "net-smtp"
 
 # Use the Defra Ruby Aws gem for loading files to AWS buckets
-gem "defra_ruby_aws", "~> 0.5"
+gem "defra_ruby_aws"
 
 # Use the waste carriers engine for the user journey
 gem "waste_carriers_engine",
-    git: "https://github.com/DEFRA/waste-carriers-engine",
-    branch: "main"
```
Comment (PaulDoyle-DEFRA) [inline Gemfile:59]: Why remove this line?

File: Gemfile [hunk 3]
```diff
@@ -72,7 +71,8 @@ gem "defra_ruby_govpay"
 # With the environment properly configured, when any app in an environment needs
 # to call Companies House, instead it will call this app which will mock the end
 # point and return the response expected.
-gem "defra_ruby_mocks"
+gem "defra_ruby_mocks",
+    git: "https://github.com/DEFRA/defra-ruby-mocks.git"
```
Comment (PaulDoyle-DEFRA) [inline Gemfile:75]: Why not use the gem version from rubygems.org?

Comment (jjromeo) [reply Gemfile:75]: Let me publish the new version and then I will use that

**Review by PaulDoyle-DEFRA — APPROVED** [review]

**Pull request merged**